File: interpreter/go/interpreter.go
package main

import (
	"fmt"
	"mop-lang/interpreter/go/internal/interpreter"
	"os"
)

func main() {
	// Check for command line arguments
	if len(os.Args) < 2 {
		fmt.Fprintf(os.Stderr, "Usage: go run interpreter.go <program_file>\n")
		os.Exit(1)
	}

	// Get program file path from arguments
	programFilepath := os.Args[1]

	// Create and run interpreter
	interp := interpreter.New(256)
	if err := interp.ExecuteProgram(programFilepath); err != nil {
		fmt.Fprintf(os.Stderr, "Error: %v\n", err)
		os.Exit(1)
	}
}

File: interpreter/python/interpreter.py
import sys
from mopl import Interpreter


def main():
    # Check for command line arguments
    if len(sys.argv) != 2:
        print("Usage: python interpreter.py <program_file>", file=sys.stderr)
        sys.exit(1)

    # Get program file path from arguments
    program_filepath = sys.argv[1]

    # Create and run interpreter
    interpreter = Interpreter()
    interpreter.execute_program(program_filepath)


if __name__ == "__main__":
    main()
